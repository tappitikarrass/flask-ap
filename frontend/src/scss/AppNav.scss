@use "base16" as *;

#app-nav {
    background-color: $base00;
    position: sticky;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    box-sizing: border-box;

    border-bottom: 1px $base03 solid;

    width: 100%;
    height: 6vh;

    & > a {
        display: flex;
        justify-content: center;
        align-items: center;

        height: 100%;
        width: 100%;

        &:hover,
        &:link,
        &:active,
        &:visited {
            color: $base07;
            text-decoration: none;
        }
        &:hover {
            background-color: $base01;
        }
    }
}

.active {
    background-color: $base01;
}
