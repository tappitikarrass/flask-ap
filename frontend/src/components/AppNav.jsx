import React from 'react';
import { NavLink } from 'react-router-dom';

import '../scss/AppNav.scss';

function AppNav() {
  return (
    <nav id="app-nav">
      <NavLink id="homepage" to="/">
        Homepage
      </NavLink>
      <NavLink id="profile" to="/profile">
        Profile
      </NavLink>
      <NavLink id="login" to="/login">
        Login
      </NavLink>
      <NavLink id="signup" to="/signup">
        Sign Up
      </NavLink>
      <NavLink id="anime" to="/anime">
        Anime
      </NavLink>
    </nav>
  );
}

export default AppNav;
