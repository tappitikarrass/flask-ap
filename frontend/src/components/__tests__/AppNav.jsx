import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import AppNav from '../AppNav';

describe('AppNav', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <AppNav />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
