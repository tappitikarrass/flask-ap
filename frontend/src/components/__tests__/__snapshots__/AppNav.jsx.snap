// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav  1`] = `
<nav
  id="app-nav"
>
  <a
    aria-current="page"
    className="active"
    href="#/"
    id="homepage"
    onClick={[Function]}
  >
    Homepage
  </a>
  <a
    className=""
    href="#/lists"
    id="user-lists"
    onClick={[Function]}
  >
    Lists
  </a>
  <a
    className=""
    href="#/admin/users"
    id="users-list"
    onClick={[Function]}
  >
    Users List
  </a>
  <a
    className=""
    href="#/profile"
    id="profile"
    onClick={[Function]}
  >
    Profile
  </a>
  <a
    className=""
    href="#/login"
    id="login"
    onClick={[Function]}
  >
    Login
  </a>
  <a
    className=""
    href="#/signup"
    id="signup"
    onClick={[Function]}
  >
    Sign Up
  </a>
</nav>
`;
